inject/src/main/java/io/micronaut/context/annotation/Requires.java

```diff
@@ -204,6 +204,25 @@
      */
     Family[] notOs() default {};
 
+    /**
+     * Expresses that the bean the given class should be available for the bean or configuration to load.
+     * Can be used in combination with {@link #beanProperty()} to specify the required bean property
+     *
+     * @return The configuration properties class
+     * @since 3.3.0
+     */
+    Class bean() default void.class;
+
+    /**
+     * Used in combination with {@link #bean()} to
+     * express that the given bean property should be
+     * set for the bean to load.
+     *
+     * @return The configuration property that should be set.
+     * @since 3.3.0
+     */
+    String beanProperty() default "";
```

Comment:
Perhaps this member is unnecessary and it could expressed as just `@Requires(bean=Foo, property="bar")` without adding a new member. Thoughts @jameskleeh ?

Reply:
In general I would agree but I think that would change the current behavior, thus making it a 4.0 thing.

`@Requires(bean=X, property="y")` today means the `X` bean has to be present and the configuration property `y` has to be present. Reusing `property` for this case changes the semantics.

Reply:
Ok leave it as is then